share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorService.java

```diff
@@ -240,9 +251,96 @@ public void startup(
 
         log.info("Starting up.");
         numPartitions = shareGroupTopicPartitionCount.getAsInt();
+        Map<TopicPartition, Long> offsets = new ConcurrentHashMap<>();
```

Comment:
1. offsets => lastPrunedOffsets? 
2. Would it be better to make that an instance val so that we don't have to pass it around?
3. Should we remove entries when `onResignation` is called?